torchmetrics/collections.py

```diff
@@ -182,16 +182,38 @@ def persistent(self, mode: bool = True) -> None:
         """Method for post-init to change if metric states should be saved to
         its state_dict
         """
-        for _, m in self.items():
+        for _, m in self.items(keep_base=True):
             m.persistent(mode)
 
     def _set_name(self, base: str) -> str:
         name = base if self.prefix is None else self.prefix + base
         name = name if self.postfix is None else name + self.postfix
         return name
 
+    def keys(self, keep_base: bool = False):
+        r"""Return an iterable of the ModuleDict key.
+        Args:
+            keep_base: Whether to add prefix/postfix on the items collection.
+        """
+        if keep_base:
+            return self._modules.keys()
+        return [self._set_name(k) for k in self._modules.keys()]
```

Comment:
I don't like that we now have a subtle difference based on the flag. By default, we'll return a dictview object (https://docs.python.org/3/library/stdtypes.html#dict-views), but if we set a flag we'll get a list instead (which can be accidentally mutated for example, etc).

What's the value of keeping old behavior? Given the prefix was provided in clone, maybe just append it at that time?

Reply:
Good catch ! I changed this.